pxtblocks/blocklyloader.ts

```diff
@@ -3199,4 +3212,18 @@ namespace pxt.blocks {
             }
         }
     }
+
+    let iconCanvasCache: Map<string> = {};
+    function iconToFieldImage(id: string): Blockly.FieldImage {
+        let url = iconCanvasCache[id];
+        return new Blockly.FieldImage(url, 30, 30, false, '');
+    }
+
+    function initJresIcons(blockInfo: pxtc.BlocksInfo) {
+        Object.keys(blockInfo.jres).forEach((jresId) => {
```

Comment:
should you clear iconCanvasCache to keep it from leaking icons between project reloads?

Reply:
Yes.